pkg/reconciler/kubernetes/tektonpipelinesascode/tektonpipelinesascode.go

```diff
@@ -0,0 +1,274 @@
+			// wait for pipeline status to change
+			return fmt.Errorf(common.PipelineNotReady)
+		}
+		// (tektonpipeline.operator.tekton.dev instance not available yet)
+		tpac.Status.MarkDependencyMissing("tekton-pipelines does not exist")
+		return err
+	}
+
+	if _, err := common.TriggerReady(r.triggerInformer); err != nil {
+		if err.Error() == common.TriggerNotReady {
+			tpac.Status.MarkDependencyInstalling("tekton-triggers is still installing")
+			// wait for trigger status to change
+			return fmt.Errorf(common.TriggerNotReady)
+		}
+		// (tektontrigger.operator.tekton.dev instance not available yet)
+		tpac.Status.MarkDependencyMissing("tekton-triggers does not exist")
+		return err
+	}
+
+	tpac.Status.MarkDependenciesInstalled()
+
+	// Pass the object through defaulting
+	tpac.SetDefaults(ctx)
+
+	if err := r.extension.PreReconcile(ctx, tpac); err != nil {
+		tpac.Status.MarkPreReconcilerFailed(err.Error())
+		return err
+	}
+
+	tpac.Status.MarkPreReconcilerComplete()
+
+	exist, err := checkIfInstallerSetExist(ctx, r.operatorClientSet, r.version, tpac)
+	if err != nil {
+		return err
+	}
+
+	if !exist {
+
+		manifest := r.manifest
+		if err := r.transform(ctx, &manifest, tpac); err != nil {
+			logger.Error("manifest transformation failed:  ", err)
+			return err
+		}
+
+		return createInstallerSet(ctx, r.operatorClientSet, tpac, r.manifest, r.version, "pac")
+	}
+
+	err = r.checkComponentStatus(ctx, tpac)
+	if err != nil {
+		tpac.Status.MarkInstallerSetNotReady(err.Error())
+		return nil
+	}
+
+	tpac.Status.MarkInstallerSetReady()
+
+	if err := r.extension.PostReconcile(ctx, tpac); err != nil {
+		tpac.Status.MarkPostReconcilerFailed(err.Error())
+		return err
+	}
+
+	tpac.Status.MarkPostReconcilerComplete()
+
+	return nil
+}
+
+func (r *Reconciler) checkComponentStatus(ctx context.Context, tpac *v1alpha1.TektonPipelinesAsCode) error {
+
+	// Check if installer set is already created
+	installerSet := tpac.Status.InstallerSet
+	if installerSet != "" {
+
+		ctIs, err := r.operatorClientSet.OperatorV1alpha1().TektonInstallerSets().
+			Get(ctx, installerSet, metav1.GetOptions{})
+		if err != nil {
+			if errors.IsNotFound(err) {
+				return nil
+			}
+			return err
+		}
+
+		ready := ctIs.Status.GetCondition(apis.ConditionReady)
+		if ready == nil || ready.Status == corev1.ConditionUnknown {
+			return fmt.Errorf("InstallerSet %s: waiting for installation", ctIs.Name)
+		} else if ready.Status == corev1.ConditionFalse {
+			return fmt.Errorf("InstallerSet %s: ", ready.Message)
+		}
+	}
+
+	return nil
+}
+
+// checkIfInstallerSetExist checks if installer set exists for a component and return true/false based on it
+// and if installer set which already exist is of older version then it deletes and return false to create a new
+// installer set
+func checkIfInstallerSetExist(ctx context.Context, oc clientset.Interface, relVersion string, tpac *v1alpha1.TektonPipelinesAsCode) (bool, error) {
```

Comment:
```suggestion
func checkIfValidInstallerSetExist(ctx context.Context, oc clientset.Interface, relVersion string, tpac *v1alpha1.TektonPipelinesAsCode) (bool, error) {
```

as this function is doing both "exists" check and "latest version" check, let us make the name proclaim it.